Cmis write

bwrsandman opened the pull request:
Port of https://code.launchpad.net/~savoirfairelinux-openerp/knowledge-addons/cmis_write/+merge/213940 by @ehdem 

re-port of https://github.com/OCA/knowledge/pull/8

**Depends on #2**

Add cmis_write: It allows to create docs from OE in DMS
## LP reviews
- @max3903: Approve on 2014-04-19
- @bwrsandman: Needs Fixing on 2014-05-06
  
  > No unittests

ehdem committed:
[ADD] add cmis_write: It allows to create docs from OE in DMS

ehdem committed:
[IMP] fix pep8 error, add control when writting from OE

ehdem committed:
[IMP] fix bug when we create job with session

ehdem committed:
[IMP] Take comments from LP

ehdem committed:
[IMP] Change the module description

ehdem committed:
[IMP] FIxed comments from LP, added unittests,fix pep8 errors

ehdem committed:
[IMP] update test file

ehdem committed:
[IMP] update the  create_doc_in_edm function

ehdem committed:
[IMP] Download document from DMS

ehdem committed:
[FIX] Fix pep8 errors

max3903 commented:
:-1: 

I get this error message in the failing job to write a document in the DMS :

Traceback (most recent call last): 
File "/home/max/openerp/cmis/git/connector/connector/queue/worker.py", line 122, in run_job job.perform(session) 
File "/home/max/openerp/cmis/git/connector/connector/queue/job.py", line 491, in perform self.result = self.func(session, _self.args, *_self.kwargs) 
File "/home/max/openerp/cmis/git/cmis/cmis_write/ir_attachment.py", line 182, in create_doc_in_edm context=session.context) 
TypeError: _auth() takes at least 4 arguments (4 given)

bwrsandman commented:
@max3903 simply pass `ids` when calling `_auth`

ehdem committed:
[FIX] Manage multi-backend

ehdem commented:
Dependency with cmis module missing. I have to add the cmis module.

bwrsandman commented:
@ehdem this is your pull request, by the way...

JordiBForgeFlow commented on cmis_write/ir_attachment.py:
```diff
@@ -0,0 +1,243 @@
+                    cr, uid, ids, ['id_dms'], context=context)[0]
+                id_dms = cmis_backend_rec['id_dms']
+                # Get results from id of document
+                results = repo.query(" SELECT * FROM  cmis:document WHERE \
```
I have tested this, and retrieves an empty list. What works instead of the query is:
result = repo.getObject(id_dms)

JordiBForgeFlow commented:
I would like to help merge this PR, but I don't know how to do it! I have download cmis_write from the launchpad repository https://code.launchpad.net/~savoirfairelinux-openerp/knowledge-addons/cmis_write, but looks like that is a slightly different version.

ehdem commented:
@jbeficent : Don't use  cmis_write from the launchpad repository https://code.launchpad.net/~savoirfairelinux-openerp/knowledge-addons/cmis_write. It's an old version

ehdem commented:
The cmis module is already in https://github.com/OCA/connector-cmis

ehdem committed:
[UPD] Improve the code

* Separate each model in distinct file
* Remove unused variables

ehdem committed:
[UPD] Update translation

ehdem committed:
[UPD] Update unittests

ehdem committed:
[UPD] Access the document and model

* Disable popup message when id_dms or DMS is not available
in the document and the model.
* Add warning message in the log when DMS is not available
or the document is delete in DMS

ehdem committed:
[UPD] Download document

Download document if one of backends is down

ehdem committed:
[UPD] Popup message display 2 times

When deleting a document, the popup message
displays two times

ehdem committed:
[FIX]Fix travis error

max3903 commented:
:+1:

ehdem committed:
[UPD] Change OpenERP term

ehdem committed:
[UPD] Add permissions for ir_attachment_doc_backend model

ehdem committed:
[UPD] Remove type field: It is deprecated

max3903 commented:
Visual check on runbot.

:+1:

dufresnedavid commented on cmis_write/ir_attachment.py:
```diff
@@ -0,0 +1,283 @@
+                      dict_metadata, user_login, filters=None):
+    """
+    This method allows to create a doc from Odoo to DMS
+    """
```
@ehdem You don't seem to use any connector unit. Why? Connector units are the core of the connector framework. Without using these, your code will be hard to maintain and inherit. Also, you don't make any use of the framework's exceptions. How will the worker know whether to retry a job or to cancel it?

ehdem replied:
@dufresnedavid : Yes, I don't use it yet. http://odoo-connector.com/guides/concepts.html#connectorunit

bwrsandman commented:
:-1: That is my issue with the connector too.

This doesn't seem to properly use the connector framework.

ehdem committed:
[UPD] Save the document from DMS

* When cmis_read is installed, the document from DMS
must be saved in Odoo.
* Fix the bug to get the name of document when downloading
from knowledge menu

ehdem committed:
[UPD] Use connector unit

Readapt the code with connector unit

ehdem committed:
[UPD] Catch error when creatin document from Odoo

Catch error when one of backend is down.
Add log message when creating document.

ehdem committed:
[FIX] Fix flake8 error

lmignon commented:
With the commit  d469e46, this PR depends of https://github.com/OCA/connector-cmis/pull/7
For you  information, the code also works on 8.0 as it is.
